src/webui/api/torrentscontroller.cpp

```diff
@@ -886,9 +886,10 @@ void TorrentsController::deleteAction()
 
     const QStringList hashes {params()["hashes"].split('|')};
     const bool deleteFiles {parseBool(params()["deleteFiles"], false)};
-    applyToTorrents(hashes, [deleteFiles](BitTorrent::TorrentHandle *const torrent)
+    const DeleteOption deleteOption = deleteFiles ? TorrentAndFiles : Torrent;
+    applyToTorrents(hashes, [deleteOption](BitTorrent::TorrentHandle *const torrent)
```

Comment:
not your code but since we are at it, I will change to `const BitTorrent::TorrentHandle *torrent`